Issue 146: Feature: add arg to find-plans that limits number of plans returned (shop classic and ESS)

mpatsift opened the pull request:
Implement #146 and add tests.

Commit:
Add plan-num-limit arg to find-plans

New plan-num-limit arg modifies number of plans returned by search when :which is set to :first or :random when set to non-default value (default is 1).

NOTE: only applies to "classic" (not explicit stack) find-plans.

* shop3/shop3.lisp
 - Add arg to find-plans.
 - Update doc string.

* shop3/decls.lisp
 - Declare *plan-num-limit*.

* shop3/planning-engine/search.lisp
 - Check *plan-num-limit* in when-done macro.

Commit:
Add plan-num-limit tests

Add plan-num-limit-tests test suite.

rpgoldman committed:
Update shop-version.lisp-expr

Bump the minor version for addition of new feature.

rpgoldman committed:
Add :PLAN-NUM-LIMIT to ESS (FIND-PLANS-STACK).

Also update the copyright on shop3.lisp.

rpgoldman committed:
Update test count.

rpgoldman reviewed (approved):
Looks good to me.  I extended to `find-plans-stack` because it was easy to copy your solution.  Waiting for the tests, but will then merge.

rpgoldman committed:
Fix type specifier bug.

rpgoldman commented:
Marked as draft until I fix the last test failure.

rpgoldman committed:
Fix missing argument.

rpgoldman committed:
Fix bug revealed by new k-plans logic.

Previously, because we were pushing onto the set of plans found, I
didn't notice that we did not re-bind `*plans-found*` as required.

rpgoldman committed:
Muffle excessive test output.

rpgoldman committed:
Fix incorrect test count.

rpgoldman committed:
Fix type declaration.

Lower bound for plan-num-limit is 1, not zero.
